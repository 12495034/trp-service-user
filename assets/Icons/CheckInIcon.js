import React from 'react'
import FontAwesome5Icon from 'react-native-vector-icons/FontAwesome5'

/**
 * Check In Icon Component, returns icon based on appointments checked in state
 */
export default function CheckInIcon(props) {
    if(props.checkedIn){
        return <FontAwesome5Icon name="user-check" color="green" size={20}/>
    } else { 
        return <FontAwesome5Icon name="user-clock" color="red" size={20}/>
    }
    
}
