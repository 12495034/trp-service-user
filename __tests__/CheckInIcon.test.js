import React from 'react';
import renderer from 'react-test-renderer';
import CheckInIcon from '../assets/Icons/CheckInIcon';

describe('Check In Icon Tests', () => {
  test('check render of checked in icon for checked in user', () => {
    const tree = renderer.create(<CheckInIcon
      checkedIn={true}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('check render of checked in icon for non checked in user', () => {
    const tree = renderer.create(<CheckInIcon
      checkedIn={false}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});