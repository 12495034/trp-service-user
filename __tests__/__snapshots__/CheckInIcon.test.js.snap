// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render of checked in icon for checked in user 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": "green",
        "fontSize": 20,
      },
      undefined,
      {
        "fontFamily": "FontAwesome5Free-Solid",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {
        "fontWeight": "900",
      },
    ]
  }
>
  
</Text>
`;

exports[`check render of checked in icon for non checked in user 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    [
      {
        "color": "red",
        "fontSize": 20,
      },
      undefined,
      {
        "fontFamily": "FontAwesome5Free-Solid",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      {
        "fontWeight": "900",
      },
    ]
  }
>
  
</Text>
`;
